File: agent/library/src/main/java/org/matilda/di/MatildaComponent.java
package org.matilda.di;

import dagger.Component;
import org.matilda.commands.CommandsModule;
import org.matilda.di.destructors.DestructionManager;
import org.matilda.messages.MessageServer;

import javax.inject.Singleton;

@Component(modules = {MatildaConnectionModule.class, MessagesModule.class, LoggerModule.class,
        CommandsModule.class, UtilsModule.class})
@Singleton
public interface MatildaComponent {
    MessageServer messageListener();

    DestructionManager destructionManager();
}

File: agent/library/src/main/java/org/matilda/di/MatildaComponent.kt
package org.matilda.di

import me.tatarka.inject.annotations.Component
import org.matilda.commands.CommandsComponent
import org.matilda.commands.MatildaScope
import org.matilda.di.destructors.DestructionManager
import org.matilda.messages.MessageServer

@Component
@MatildaScope
abstract class MatildaComponent(@Component val connectionComponent: MatildaConnectionComponent,
                                @Component val loggerComponent: LoggerComponent) :
    MessagesComponent, CommandsComponent, UtilsComponent {
    abstract val messageListener: MessageServer
    abstract val destructionManager: DestructionManager
}
